### PR TITLE
[Assistnats] improve reporting

### PR DESCRIPTION
**Pull request by mishig25**
Post to webhook url "report assistant" events

### CODE REVIEW

File: src/routes/settings/assistants/[assistantId]/+page.server.ts [hunk 1]
```diff
@@ -71,6 +73,31 @@ export const actions: Actions = {
 		if (!acknowledged) {
 			return fail(500, { error: true, message: "Failed to report assistant" });
 		}
+
+		if (SLACK_URL_REPORT_ASSISTANT) {
+			const prefixUrl = PUBLIC_SHARE_PREFIX || `${PUBLIC_ORIGIN || url.origin}${base}`;
+			const assistantUrl = `${prefixUrl}/assistant/${params.assistantId}`;
+
+			const assisntant = await collections.assistants.findOne(
```
Comment (nsarrazin) [inline src/routes/settings/assistants/[assistantId]/+page.server.ts:81]:
```suggestion
			const assistant = await collections.assistants.findOne(
```

File: .env [hunk 2]
```diff
@@ -133,4 +133,6 @@ EXPOSE_API=true
 
 ENABLE_ASSISTANTS=false #set to true to enable assistants feature
 
-ALTERNATIVE_REDIRECT_URLS=`[]` #valide alternative redirect URL for OAuth
\ No newline at end of file
+ALTERNATIVE_REDIRECT_URLS=`[]` #valide alternative redirect URL for OAuth
+
+SLACK_URL_REPORT_ASSISTANT=#provide https://api.slack.com/messaging/webhooks to get notified when an assistant gets reported
```
Comment (nsarrazin) [inline .env:138]: Is this a secret or will it be in the public  `.env.template` ?

**Review by nsarrazin — APPROVED** [review]
Looks good to me can't test the implem itself but code looks good! left a simple typo fix

EDIT: Tested locally with a test webhook and it works great

File: src/routes/settings/assistants/[assistantId]/+page.server.ts (hunk 1 above)
Comment (mishig25) [reply src/routes/settings/assistants/[assistantId]/+page.server.ts]: handled in 617461553aacfd469a8dc5cc37ab24d15af92ab6

File: src/routes/settings/assistants/[assistantId]/+page.server.ts [hunk 3]
```diff
@@ -71,6 +73,31 @@ export const actions: Actions = {
...
+				{ projection: { name: 1 } }
+			);
+
+			const res = await fetch(SLACK_URL_REPORT_ASSISTANT, {
+				method: "POST",
+				headers: {
+					"Content-type": "application/json",
+				},
+				body: JSON.stringify({
+					text: `Assistant <${assistantUrl}|${assisntant?.name}> reported by <http://hf.co/${locals.user?.username}|${locals.user?.username}>`,
```
[the remark this answers is not in the document]
Comment (mishig25) [reply src/routes/settings/assistants/[assistantId]/+page.server.ts]: handled in 617461553aacfd469a8dc5cc37ab24d15af92ab6

File: src/routes/settings/assistants/[assistantId]/+page.server.ts [hunk 4]
```diff
@@ -71,6 +73,31 @@ export const actions: Actions = {
...
+				{ projection: { name: 1 } }
+			);
+
+			const res = await fetch(WEBHOOK_URL_REPORT_ASSISTANT, {
+				method: "POST",
+				headers: {
+					"Content-type": "application/json",
+				},
+				body: JSON.stringify({
+					text: `Assistant <${assistantUrl}|${assistant?.name}> reported by <http://hf.co/${locals.user?.username}|${locals.user?.username}>`,
```
Comment (mishig25) [inline src/routes/settings/assistants/[assistantId]/+page.server.ts:92]: [this is slack msg template for links](https://api.slack.com/reference/surfaces/formatting#linking-urls)

File: .env (hunk 2 above)
Comment (nsarrazin) [reply .env]: First you'll need to add it [here](https://github.com/huggingface/chat-ui/settings/secrets/actions) on github

Then you'll need to update [the script that generates the prod config file](https://github.com/huggingface/chat-ui/blob/main/scripts/updateProdEnv.ts) should be fairly straight forward, just get `process.env.WEBHOOK_URL_REPORT_ASSISTANT` and add it to `full_config` 


Then optionally for local testing you can add `WEBHOOK_URL_REPORT_ASSISTANT=<here>` to your `.env.SECRET_CONFIG` but it's not a requirement. (only useful if you make use of `npm run updateLocalEnv` )

**Review by coyotte508 — APPROVED** [review]

File: .env (hunk 2 above)
Comment (mishig25) [reply .env]: > First you'll need to add it [here](https://github.com/huggingface/chat-ui/settings/secrets/actions) on github

done

> Then you'll need to update [the script that generates the prod config file](https://github.com/huggingface/chat-ui/blob/main/scripts/updateProdEnv.ts) should be fairly straight forward, just get process.env.WEBHOOK_URL_REPORT_ASSISTANT and add it to full_config

done 729789c1a5b213b1df9f8054551b20534dcdc35f

**Review by nsarrazin — APPROVED** [review]

File: scripts/updateProdEnv.ts [hunk 5]
```diff
@@ -6,6 +6,7 @@ const SERPER_API_KEY = process.env.SERPER_API_KEY;
 const OPENID_CONFIG = process.env.OPENID_CONFIG;
 const MONGODB_URL = process.env.MONGODB_URL;
 const HF_TOKEN = process.env.HF_TOKEN ?? process.env.HF_ACCESS_TOKEN; // token used for API requests in prod
+const WEBHOOK_URL_REPORT_ASSISTANT = process.env.WEBHOOK_URL_REPORT_ASSISTANT; // slack webhook url used to get reported assistants
```
Comment (mishig25) [inline scripts/updateProdEnv.ts:9]:
```suggestion
const WEBHOOK_URL_REPORT_ASSISTANT = process.env.WEBHOOK_URL_REPORT_ASSISTANT; // slack webhook url used to get "report assistant" events
```

**Pull request merged**